### DISCUSSION TITLE
Error when run debug windows

### DISCUSSION THREAD

**Discussion by NgVThangBz** [axmolengine/axmol#2295] [Help Desk]
- axmol version: 2.2.1
- devices test on: windows 10
- developing environments
   - NDK version: r23c
   - Visual Studio: 
      - VS version: 2022 (17.12.3)
      - MSVC version: 19.39+
      - Windows SDK version: 10.0.22621.0+
   - cmake version: 3.31.2
Steps to Reproduce:
- setup.ps1
- axmol build
- run ok in release
- in debug i received the error(0xc000007b):
![image](https://github.com/user-attachments/assets/fe247adf-7cd5-4cf4-b91c-3c7965831629)

**Comment by halx99** [comment]
two possible reason:

- directx runtime not ready
- vc runtime incorrect: system32 folder contains 32bit runtime dll or SysWow64 contains 64bit runtime dll

**Comment by NgVThangBz** [comment]
@halx99 so, how fix it?
delete vcruntime140d.dll?

**Comment by NgVThangBz** [comment]
this log:
![image](https://github.com/user-attachments/assets/747ff0f3-80f5-4086-94c3-faa923450824)

**Comment by halx99** [comment]
try uninstall and reinstall vcredist

**Comment by NgVThangBz** [comment]
it doesn't seem to work for me

**Comment by NgVThangBz** [comment]
- uninstall and reinstall vcredist ( x86 and x64)
- run DirectX again
but not work, has any ideal?

**Comment by rh101** [comment]
Just out of curiosity, if you put a breakpoint on `_tWinMain()` (or `main(int, char**)`, depending on which one is active) in `proj.win32/main.cpp`, does it actually hit the breakpoint before it crashes?

**Comment by NgVThangBz** [comment]
@rh101 it not come to breakpoint
![image](https://github.com/user-attachments/assets/1c9bdb80-bfb0-4126-b219-7ad3e8da1be1)

**Comment by rh101** [comment]
Is this something that just started occurring, and if so, have you checked the most recent changes you made to the application to see if that is what caused the problem?

Have you also tried deleting the build folder and regenerating it with the `axmol build` command?

**Comment by NgVThangBz** [comment]
I tried deleting the build folder and using` axmol build` many times, but the problem is still not solved.
maybe the problem lies in my dll files, but i cant figure out exactly why

**Comment by rh101** [comment]
Can you try a new project? Create one via `axmol new` etc., and see if the debug build works for that.

**Comment by NgVThangBz** [comment]
i also tried creating new, same error.
maybe it's the environment(dll) of the device i'm using

**Comment by rh101** [comment]
> maybe it's the environment(dll) of the device i'm using

What do you mean by this remark?

Is this a local PC with Visual Studio 2022 installed? Have you tried repairing the VS installation?

**Comment by NgVThangBz** [comment]
> What do you mean by this remark?

@rh101 
i think have some dll file incorect

and
I have tried other types:
[x64-debug] not work
[x64-release] work
[win32-debug] work
[win32-release] work

**Comment by halx99** [comment]
Try use `wsls-arch` to detect all dlls bitness in folder: `C:\Windows\system32`:

https://github.com/simdsoft/wsls/releases/download/v3.5.3.1/wsls-3.5.3.1.7z

**Comment by NgVThangBz** [comment]
> Try use `wsls-arch` to detect all dlls bitness in folder: `C:\Windows\system32`:
> 
> https://github.com/simdsoft/wsls/releases/download/v3.5.3.1/wsls-3.5.3.1.7z

How to set it up?
i try open file.exe and bat?

**Comment by halx99** [comment]
the package contains `wsls-arch.exe` which is a independent program, only accept one parameter to specify dll file path for detecting and will print x86 or x64.

**Comment by halx99** [comment]
you can write a .bat or .ps1 to iterate dlls in `C:\Windows\System32\` and invoke `wsls-arch.exe`

**Comment by NgVThangBz** [comment]
The problem is still not solved, however I think this is not axmol's problem, so I will close this issue,
 if necessary I will ask for everyone's opinion in the Discussion
thank for all

**Comment by Ts-pe** [comment]
Aren't you trying to “Axmol build” inside the engine folder? (it is not very clear where you are doing the axmol build, but from the name of the error window it looks like you are doing it in a new project and not in the engine).

Try this command “cmake -S . -B build -G “Visual Studio 17 2022” -A x64” (in a new project), open the solution (.sln), it will default to debug/ x64. just hit start debugging (it will take a while).

PS: you need to delete everything you have done with your engine version (reinstall axmol), another option would be to upgrade the engine (you can always go back to a previous version by reinstalling axmol, but you must be careful that the folder is always named the same or you will have many axmol folders in your environment variables, although it is not difficult to remove them, just extra work).
